### PR TITLE
Minor bugfixes and added Dockerfile

### PR DESCRIPTION
**Pull request by jojo141185**

### CODE REVIEW

**Review by dougy147 — APPROVED** [review]
Thanks for your contribution.
Would you mind consider reviewing my steps in the README for Docker instructions?

File: func/08_mac [hunk 1]
```diff
@@ -10,7 +10,7 @@ random_mac() {
 	else
 		mac_prefix=${default_mac_prefix}
 	fi
-	number_chars_prefix=$(( $(echo ${mac_prefix} | sed 's/://g' | wc --chars) - 1 ))
```
[the remark this answers is not in the document]
Comment (jojo141185) [reply func/08_mac]: Yes, but the "-m" argument for wc is much more compatible. For example, if your operating system relies on [busybox](https://busybox.net/BusyBox.html), “--chars” is not currently supported.

**Comment by jojo141185** [comment]
> Thanks for your contribution. Would you mind consider reviewing my steps in the README for Docker instructions?

That looks good! Thank's for adding.

**Comment by jojo141185** [comment]
I don't want to open a new PR, but I think the following [regexp](https://github.com/dougy147/mcbash/blob/f8fc4f795cd2f60e5875c8f6db5add063fb00a31/bin/mcbash#L157-L158) in your last code update is broken:
```
        # Check if DNS is well formed URL or IP
        if ! [[ $(echo "${1}" | grep -Eo '((https?|ftp)://)?([a-Z0-9-]+\.)?[a-Z0-9-]+\.[a-Z0-9-]+((:)[0-9]{1,5})?(/.*)?') == "${1}" ||
```
This one should work:
((https?|ftp):\/\/)?([a-zA-Z0-9-]+\.)?[a-zA-Z0-9-]+\.[a-zA-Z0-9-]+((:)[0-9]{1,5})?(\/.*)?

Additionally, you should check whether the current proxy server used is working. Some proxies do not time out. They return an error on the handshake and mcbash assumes the request worked and just continues...

```
---------------------------------
        Handshake token : error code: 1001
[26] 00:1A:79:EF:17:8D 
---------------------------------
        Handshake token : error code: 1001
[27] 00:1A:79:EF:17:8E 
```

In this case it should rotate to the next proxy server in the list.

**Comment by dougy147** [comment]
I've changed the regexp to `((https?|ftp)://)?([-a-zA-Z0-9]+\.)?[-a-zA-Z0-9]+\.[-a-zA-Z0-9]+((:)[0-9]{1,5})?(/.*)?` (the dash should be placed first in a class of characters, also not sure if `a-Z` == `a-zA-Z` but let's go for the latter).

> Additionally, you should check whether the current proxy server used is working. Some proxies do not time out. They return an error on the handshake and mcbash assumes the request worked and just continues...

Yes, I've been decomposing the handshake function yesterday to facilitate those kind of adjustements, I will soon take a look at requests success/failure handling.

> I don't want to open a new PR, but [...]

PR will not bother me ;)

**Comment by jojo141185** [comment]
Thank's for updating. it's working now! With the previous regexp I received an "ERROR: invalid DNS format.". 
The regexp range [a-Z] matches any character that falls between 'a' and 'Z' in the ASCII character set, which includes some non-alphabetical characters including '[' (91), '\' (92), ']' (93), '^' (94), '_' (95), and '`' (96). Accordingly, [a-zA-Z] is more accurate and what we need here.

**Pull request merged**